WebIDL/ecmascript-binding/legacy-platform-object.html

```diff
@@ -188,4 +192,20 @@
                           {value: option0, writable: true, enumerable: true,
                            configurable: true});
 }, "Test [[DefineOwnProperty]] with indexed property setter support.");
+
+test(function() {
+    // https://heycam.github.io/webidl/#is-an-array-index
+    assert_equals(document.all["4892.123"].id, "4892.123",
+        "Numbers with decimal points are not treated as array indices");
+    assert_equals(document.all["-0"], undefined, "'-0' is special cased");
```

Comment:
This is tested by https://github.com/w3c/web-platform-tests/pull/9303. Testing some other object that `document.all` would be nice.